File: rgctl-tests/ecommerce-c/src/cfg_features.c
/* CFG feature probes for rgctl expected-facts (C lowering coverage). */

int cfg_short_circuit(int a, int b) {
    if (a && b) {
        return 1;
    }
    return 0;
}

int cfg_do_while(int n) {
    int i = 0;
    int total = 0;
    do {
        total += i;
        i++;
    } while (i < n);
    return total;
}

int cfg_goto_label(int x) {
    if (x < 0) {
        goto done;
    }
    x++;
done:
    return x;
}

int cfg_switch_fallthrough(int x) {
    int r = 0;
    switch (x) {
    case 1:
        r += 1;
        /* fall through */
    case 2:
        r += 2;
        break;
    default:
        r = 0;
        break;
    }
    return r;
}

int cfg_ternary(int x) {
    return x > 0 ? x : -x;
}

File: rgctl-tests/ecommerce-rust/src/cfg_features.rs
//! CFG feature probes for rgctl expected-facts (Rust lowering coverage).

#![allow(dead_code)]

pub fn cfg_short_circuit(a: bool, b: bool) -> i32 {
    if a && b {
        1
    } else {
        0
    }
}

pub fn cfg_match_guard(x: i32) -> i32 {
    match x {
        n if n > 0 => n,
        _ => 0,
    }
}

pub fn cfg_try_op(x: Result<i32, ()>) -> Result<i32, ()> {
    let v = x?;
    Ok(v + 1)
}

pub fn cfg_for_in(xs: &[i32]) -> i32 {
    let mut total = 0;
    for v in xs {
        total += *v;
    }
    total
}

pub async fn cfg_await() -> i32 {
    let x = async { 1 }.await;
    x
}
